draft-ietf-tls-esni.md

```diff
@@ -756,10 +756,11 @@ ClientHello message as follows:
 
 1. It constructs EncodedClientHelloInner as described in {{encoding-inner}}.
 
-1. It constructs a second partial ClientHelloOuterAAD message. This message MUST
-   be syntactically valid. The extensions MAY be copied from the original
-   ClientHelloOuter unmodified, or omitted. If not sensitive, the client MAY
-   copy updated extensions from the second ClientHelloInner for compression.
+1. It constructs a second partial ClientHelloOuterAAD. This MUST be constructed
+   afresh as described in {{encrypting-clienthello}}. The extensions MAY be copied
```

Comment:
This section doesn't even talk about constructing ClientHelloOuterAAD. Did you mean to cite something else? Though I'm not sure there is anything useful to cite here. I avoided citing {{real-ech}} when reorganizing this because that contains a ton of text specific to the first ClientHello.

"MUST be constructed afresh" also does not seem particularly meaningful or accurate. Ultimately, the ClientHelloOuter wrapper is entirely pointless here. Both sides know it will be ignored. Some clients will find it easiest to apply RFC8446 rules (reuse some code), while others cannot because whether HelloRetryRequest will consistently apply to both depends on complex rules we never managed to write down. (This was the crux of the HelloRetryRequest fuss.) Since we didn't want to define a whole new message without all the extra baggage, this section just says to do whatever. And then maybe we'll learn something about interop needs during experimentation.

Reply:
So what was the intent of the original sentence? Maybe just deleting it would be fine, depending.

Reply:
It probably can be dropped. It was mostly just "the other side should be able to parse it". We're in this weird situation where we're producing a ClientHelloOuter2 that may well not be consistent with ClientHelloOuter1 + HelloRetryRequest. But both sides have agreed it's meaningless... we're only sending it because I guess we haven't defined a better message.

There is probably a less goofy design buried in here somewhere. :-)

Reply:
I propose we park this for now and file an issue to ask ourselves if we want a new message for this second CHOuter. @davidben or @sftcd, would one of you mind filing that issue?

Reply:
Ah - the "both sides have agreed it's meaningless" finally enlightened me - what the original odd MUST meant is "the client hello outer2 is mostly useless, but MUST be well formed even so". I'm fine if we park that, just delete the odd sentence or replace with whatever the editors' think better.

Reply:
Parking SGTM. I am also okay deleting the sentence. I agree with you it's not really meaningful. I also agree this whole situation is confusing and silly. :-)